tests/unittests/CoreGraphics.drawing/CGContextDrawing_DrawingOptionsTests.cpp

```diff
@@ -0,0 +1,169 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include "DrawingTest.h"
+
+static CGLineCap lineCaps[] = { kCGLineCapButt, kCGLineCapSquare, kCGLineCapRound };
+static CGLineJoin lineJoins[] = { kCGLineJoinBevel, kCGLineJoinRound };
+static CGFloat miterLimits[] = { 1.5, 1.8, 2, 3 };
+struct PhaseDashCount {
+    CGFloat phase;
+    const CGFloat dashPattern[4];
+    size_t count;
```

Comment:
very confused.

if you use an `std::vector` here you don't need to cap it at 4, and count can be automatically inferred. It's infinitely better!